base/readiness_test.go

```diff
@@ -0,0 +1,166 @@
+}
+
+// TestWithoutCondition tests the task succeeds when there are no conditions on the object
+// It should be successful
+// Also validates parsing of timeout
+// Also validates setting of default namespace
+func TestWithoutConditions(t *testing.T) {
+	ns, nm := "default", "test-pod"
+	pod := newPod(ns, nm).Build()
+	rTask := NewReadinessTask(nm).WithResource("pods").WithTimeout("20s").Build()
+	runTaskTest(t, rTask, true, ns, pod)
+}
+
+// TestWithCondition tests that the task succeeds when the condition is present and True
+func TestWithCondition(t *testing.T) {
+	ns, nm := "default", "test-pod"
+	pod := newPod(ns, nm).WithCondition("Ready", "True").Build()
+	rTask := NewReadinessTask(nm).WithResource("pods").WithNamespace(ns).WithCondition("Ready").Build()
+	runTaskTest(t, rTask, true, ns, pod)
+}
+
+// TestWithFalseCondition tests that the task fails when the condition is present and not True
+func TestWithFalseCondition(t *testing.T) {
+	ns, nm := "default", "test-pod"
+	pod := newPod(ns, nm).WithCondition("Ready", "False").Build()
+	rTask := NewReadinessTask(nm).WithResource("pods").WithNamespace(ns).WithCondition("Ready").Build()
+	runTaskTest(t, rTask, false, ns, pod)
+}
+
+// TestConditionNotPresent tests that the task fails when the condition is not present (but others are)
+func TestConditionNotPresent(t *testing.T) {
+	ns, nm := "default", "test-pod"
+	pod := newPod(ns, nm).Build()
+	rTask := NewReadinessTask(nm).WithResource("pods").WithNamespace(ns).WithCondition("NotPresent").Build()
+	runTaskTest(t, rTask, false, ns, pod)
+}
+
+// TestInvalidTimeout tests that the task fails when the specified timeout is invalid (not parseable)
+func TestInvalidTimeout(t *testing.T) {
+	ns, nm := "default", "test-pod"
+	pod := newPod(ns, nm).WithCondition("Ready", "True").Build()
+	rTask := NewReadinessTask(nm).WithResource("pods").WithNamespace(ns).WithTimeout("timeout").Build()
+	runTaskTest(t, rTask, false, ns, pod)
+}
+
+// UTILITY METHODS for all tests
+
+// runTaskTest creates fake cluster with pod and runs rTask
+func runTaskTest(t *testing.T, rTask *readinessTask, success bool, ns string, pod *unstructured.Unstructured) {
+	*kd = *NewFakeKubeDriver(cli.New())
+	rs := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+	_, err := kd.DynamicClient.Resource(rs).Namespace(ns).Create(context.Background(), pod, metav1.CreateOptions{})
+	assert.NoError(t, err, "get failed")
+
+	err = rTask.run(&Experiment{
+		Tasks:  []Task{rTask},
+		Result: &ExperimentResult{},
+	})
+	if success {
+		assert.NoError(t, err)
+	} else {
+		assert.Error(t, err)
+	}
+}
+
+type podBuilder corev1.Pod
+
+func newPod(ns string, nm string) *podBuilder {
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nm,
+			Namespace: ns,
+		},
+	}
+	return (*podBuilder)(pod)
+}
+
+func (p *podBuilder) Build() *unstructured.Unstructured {
```

Comment:
Could we please not externalize any test helper methods? like Build or WithCondition ... 
***
Limiting to package local scope is better (I realize podBuilder itself is package local; same idea).

Reply:
done